sdk/src/main/java/com/mapbox/maps/MapProvider.kt

```diff
@@ -50,7 +52,10 @@ internal object MapProvider {
         paramsProvider(context, accessToken, MapboxModuleType.MapTelemetry)
       }
     }
-    mapTelemetry.onAppUserTurnstileEvent()
+    // Report turnstile event in background thread.
+    thread {
```

Comment:
Will it create a new thread on every map instantiation?

Reply:
I'd probably cached created thread together with `mapTelemetry`

Reply:
@paulrenn67 can we validate the message is sent correctly? Most of our native APIs requires to be ran on main thread only. 

cc @jush do you know the thready safety for the `EventsService.sendTurnstileEvent` API?

Reply:
@jush I checked with Tanja and also checked with added http logger, seemed to work alright.
@yunikkk I don't think it is worth storing the thread because most apps only contain a single `MapView`, and once the event has been sent, we don't need the thread anymore.

Reply:
Unfortunately, I'm not familiar enough with the native side to say for sure but it seems to enqueue it immediately to a worker...

Which makes me wonder why this call is expensive in the first place...

Reply:
Perhaps it is the lazy initialisation of `mapTelemetry = MapboxModuleProvider.createModule` that costs time, the first time through.  In which case we could maybe use the thread technique only there.

Reply:
looking at the Jira ticket Peng mentions that it's due to JSON conversion

Reply:
JSON is very small in this case as far as I have seen